### PR TITLE
[InterFlux] Consolidation polling loop

### PR DESCRIPTION
**Pull request by zeptin**
The client calls have been tested in isolation, but not as part of the full poller yet.

### CODE REVIEW

File: src/Stratis.Bitcoin/Stratis.Bitcoin.csproj [hunk 1]
```diff
@@ -20,6 +20,12 @@
     <Authors>Stratis Group Ltd.</Authors>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Stratis.Bitcoin.csproj:23]: This weird folder has been in the solution for a long time, I can remove this from the PR if necessary but I'm not sure what the folder is actually for, if anything.

**Review by fassadlr — APPROVED** [review]

**Comment by fassadlr** [comment]
Re-opened

File: src/Stratis.Bitcoin.Features.Interop/InteropPoller.cs [hunk 2]
```diff
@@ -54,6 +56,9 @@ public sealed class InteropPoller : IDisposable
         /// <summary>The number of blocks deep a submission transaction needs to be before it should start getting confirmed by the non-originating nodes.</summary>
         public readonly BigInteger SubmissionConfirmationThreshold = 12;
 
+        /// <summary>Once the node's personal Cirrus rewards wallet has more UTXOs than this, an automatic consolidation will be triggered.</summary>
+        public const int UtxoCountThreshold = 500;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Interop/InteropPoller.cs:60]: Please change to 100.

File: src/Stratis.Bitcoin.Features.Interop/CirrusContractClient.cs [hunk 3]
```diff
@@ -304,6 +309,57 @@ public async Task<TransactionVerboseModel> GetRawTransactionAsync(string transac
             return response;
         }
 
+        public async Task<WalletStatsModel> GetWalletStatsAsync(string walletName, string accountName, int minConfirmations = 1, bool verbose = false)
+        {
+            WalletStatsModel response = await this.cirrusInteropSettings.CirrusClientUrl
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Interop/CirrusContractClient.cs:314]: We will need to pass higher time outs to these methods e.g

```
                using (CancellationTokenSource cancellation = new CancellationTokenSource(TimeSpan.FromSeconds(180)))
                {
                    response = await this.cirrusInteropSettings.CirrusClientUrl
                    .AppendPathSegment("api/smartcontracts/build-and-send-call")
                    .PostJsonAsync(request, cancellation.Token)
                    .ReceiveJson<BuildCallContractTransactionResponse>()
                    .ConfigureAwait(false);
```

File: src/Stratis.Bitcoin.Features.Interop/CirrusContractClient.cs [hunk 4]
```diff
@@ -304,6 +309,57 @@ public async Task<TransactionVerboseModel> GetRawTransactionAsync(string transac
...
+                {
+                    WalletName = walletName,
+                    AccountName = accountName,
+                    WalletPassword = walletPassword,
+                    DestinationAddress = this.cirrusInteropSettings.CirrusSmartContractActiveAddress,
+                    UtxoValueThreshold = utxoValueThreshold.ToString(),
+                    Broadcast = broadcast
+                };
+
+                IFlurlResponse response = await this.cirrusInteropSettings.CirrusClientUrl
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Interop/CirrusContractClient.cs:340]: Name here, this might take up to 5 minutes depending.

**Review by fassadlr — APPROVED** [review]

**Pull request merged**